File: LeetCode/Easy/0101-symmetric-tree/0101-symmetric-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    var leftSide []int
    var rightSide []int

    inOrderLeftSide(&leftSide, root, root, 0) 
    inOrderRightSide(&rightSide, root, root, 0)
    fmt.Println(leftSide)
    fmt.Println(rightSide)
    return slices.Equal(leftSide, rightSide)
}

func inOrderLeftSide(arr *[]int, node *TreeNode, root *TreeNode, cnt int) {
    if (node == nil) {
        return
    }

    inOrderLeftSide(arr, node.Left, root, cnt+1)
    // visit
    *arr = append(*arr, cnt)
    *arr = append(*arr, node.Val)
    if (node == root) {
        return
    }
    
    inOrderLeftSide(arr, node.Right, root, cnt+1)
}

func inOrderRightSide(arr *[]int, node *TreeNode, root *TreeNode, cnt int) {
    if (node == nil) {
        return
    }
    inOrderRightSide(arr, node.Right, root, cnt+1)
    // visit
    *arr = append(*arr, cnt)
    *arr = append(*arr, node.Val)
    if (node == root) {
        return
    }
    inOrderRightSide(arr, node.Left, root, cnt+1)

}

File: LeetCode/Easy/0101-symmetric-tree/0101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        return isMirror(root.left, root.right);
    }

    public boolean isMirror(TreeNode leftNode, TreeNode rightNode) {
        if (leftNode == null || rightNode == null) {
            return leftNode == rightNode;
        }
        return leftNode.val == rightNode.val && isMirror(leftNode.left, rightNode.right) 
        && isMirror(leftNode.right, rightNode.left);
    }
}